Tests/UnitTests/Examples/Io/Csv/MeasurementReaderWriterTests.cpp

```diff
@@ -0,0 +1,179 @@
+// This file is part of the Acts project.
+//
+// Copyright (C) 2023 CERN for the benefit of the Acts project
+//
+// This Source Code Form is subject to the terms of the Mozilla Public
+// License, v. 2.0. If a copy of the MPL was not distributed with this
+// file, You can obtain one at http://mozilla.org/MPL/2.0/.
+
+#include <boost/test/unit_test.hpp>
+
+#include "Acts/Definitions/TrackParametrization.hpp"
+#include "Acts/Tests/CommonHelpers/FloatComparisons.hpp"
+#include "Acts/Tests/CommonHelpers/WhiteBoardUtilities.hpp"
+#include "Acts/Utilities/BinUtility.hpp"
+#include "Acts/Utilities/Zip.hpp"
+#include "ActsExamples/Digitization/DigitizationConfig.hpp"
+#include "ActsExamples/Digitization/Smearers.hpp"
+#include "ActsExamples/EventData/Cluster.hpp"
+#include "ActsExamples/EventData/Measurement.hpp"
+#include "ActsExamples/Io/Csv/CsvMeasurementReader.hpp"
+#include "ActsExamples/Io/Csv/CsvMeasurementWriter.hpp"
+
+#include <fstream>
+#include <iostream>
+#include <random>
+
+using namespace ActsExamples;
+using namespace Acts::Test;
+
+BOOST_AUTO_TEST_CASE(CsvMeasurmentRoundTrip) {
+  IndexSourceLinkContainer sourceLinksOriginal;
+  MeasurementContainer measOriginal;
+  ClusterContainer clusterOriginal;
+  IndexMultimap<Index> mapOriginal;
+
+  ////////////////////////////
+  // Create some dummy data //
+  ////////////////////////////
+  const std::size_t nMeasurements = 3;
+  Acts::GeometryIdentifier someGeoId{298453};
+
+  std::mt19937 gen(23);
+  std::uniform_int_distribution<unsigned> disti(1, 10);
+  std::uniform_real_distribution<double> distf(0.0, 1.0);
+
+  for (auto i = 0ul; i < nMeasurements; ++i) {
+    IndexSourceLink sl(someGeoId, static_cast<Index>(i));
+    sourceLinksOriginal.insert(sl);
+
+    Acts::Vector2 p = Acts::Vector2::Random();
+    Acts::SymMatrix2 c = Acts::SymMatrix2::Random();
+
+    auto m = Acts::makeMeasurement(Acts::SourceLink{sl}, p, c, Acts::eBoundLoc0,
+                                   Acts::eBoundLoc1);
+
+    // TODO this fails!
+    // auto m = Acts::makeMeasurement(Acts::SourceLink{sl}, p, c,
+    // Acts::eBoundLoc0,
+    //                                Acts::eBoundTime);
```

Comment:
is this still relevant?

Reply:
Hmm yeah it is not supported to write/read non-consecutive measurments currently... I don't plan to implement this at the moment...